### PR TITLE
Updated chunking scripts to reflect .csv file loading

### LINKED ISSUE #27: Chunking methods and documentation outdated for current pipeline
The new dataloader means that we no longer need to create 5-second chunks for training. Thus, chunking methods and their documentation should be updated to reflect this. I think it's a good idea to keep functionality to make these chunks since it's easier for PyTorch to load it in, but this should be de-emphasized.

### LINKED ISSUE #44: Refactor WTS_Chunking
There are some irrelevant functions in the `WTS_chunking.py` file and `dynamic_yan_chunking` requires a docstring. 

Questions on whether to keep these annotation post-processing PyHa functions in this repository or plan to import them from PyHa-Lite. While keeping them in this repository would be nice for versioning, they will eventually be merged into PyHa-Lite AND `generate_automated_labels` is still required from PyHa.

Additionally, to make this more modular, read the .csvs in `__main__` and operate on the DataFrame only.

### PR DESCRIPTION
**Pull request by sprestrelski**
Removes references to loading from 5s chunks in favor of loading audio directly from the .csv to save storage.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -188,6 +188,7 @@ good-names=i,
            df,
            a,
            b,
+           e
```
Comment (Sean1572) [inline .pylintrc:191]: What was e being used for lol

File: README.md [hunk 2]
```diff
@@ -32,52 +32,50 @@ We recommend that you use miniconda for all package management. Once miniconda i
 conda env create -f environment.yml
 ```
 
-In order to recreate our results, [PyHa](https://github.com/UCSD-E4E/PyHa) needs to be installed and set up. Furthermore, our results are based off of the [BirdCLEF2023 dataset](https://www.kaggle.com/competitions/birdclef-2023). You may also find it useful to use a  [no-call dataset](https://www.kaggle.com/code/sprestrelski/birdclef23-uniform-no-call-sound-chunks) compiled from previous competitions.
+In order to recreate our results, [PyHa](https://github.com/UCSD-E4E/PyHa) needs to be installed and set up. A sample dataset to use this repository on is the [BirdCLEF2023 dataset](https://www.kaggle.com/competitions/birdclef-2023). You may also find it useful to use a [no-call dataset](https://www.kaggle.com/code/sprestrelski/birdclef23-uniform-no-call-sound-chunks) compiled from previous competitions.
```
Comment (Sean1572) [inline README.md:35]: We should add no bird to training lowkey

File: chunking_methods/WTS_chunking.py [hunk 3]
```diff
@@ -237,11 +237,11 @@ def convolving_chunk(row, chunk_duration=3, only_slide=False):
...
     """TODO
     """
-    return_df = pd.DataFrame(columns=df.columns)
+    return_dfs = []
     for _, row in df.iterrows():
         chunk_df = convolving_chunk(row, chunk_duration=chunk_duration, only_slide=only_slide)
-        return_df = pd.concat([return_df, chunk_df],  ignore_index=True)
-    return return_df
+        return_dfs.append(chunk_df)
+    return pd.concat(return_dfs,  ignore_index=True)
```
Comment (Sean1572) [inline chunking_methods/WTS_chunking.py:244]: Good change, why I'm I going so in depth into this

File: chunking_methods/WTS_chunking.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-"""Chunking script form PyHa to convert weak labels to strong labels.
+"""Chunking script from PyHa to convert weak labels to strong labels.
```
Comment (Sean1572) [inline chunking_methods/WTS_chunking.py:1]: This was me huh

File: chunking_methods/gen_csv_labels.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+from PyHa.IsoAutio import generate_automated_labels
+from WTS_chunking import dynamic_yan_chunking
+
+ISOLATION_PARAMETERS = {
+    "model" : "tweetynet",
+    "tweety_output": True,
+    "verbose" : True
+}
+
+CHUNK_DURATION = 5
```
Comment (Sean1572) [inline chunking_methods/gen_csv_labels.py]: create config later?

File: chunking_methods/gen_csv_labels.py [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+SLIDING_CHUNKS = False
+
+# input weak annotations
+METADATA_PATH = "~/metadata.csv"
+# output for strong annotations
+STRONG_LABELS_CSV = "~/132PeruXC_Labels.csv"
+# output for chunked annotations
+CHUNKED_CSV = "~/132PeruXC_RawChunks.csv"
+
+def convert2wav(path):
```
Comment (Sean1572) [inline chunking_methods/gen_csv_labels.py]: No error message for bad extension, recommend creating a runtime error for this instead of silent return

File: chunking_methods/gen_csv_labels.py [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        subfolders.sort()
+        for s in subfolders:
+            convert2wav(path)
+            temp_df = generate_automated_labels(s, ISOLATION_PARAMETERS)
+            if temp_df.empty:
+                continue
+            automated_df = pd.concat([automated_df, temp_df], ignore_index=True, sort=False)
+    
+    if automated_df.empty:
+        print("no labels generated")
```
Comment (Sean1572) [inline chunking_methods/gen_csv_labels.py:75]: maybe "cannot automatically determine classes" and add a potential case for finding it from XC filenames?

**Review by Sean1572 — DISMISSED** [review]
Wrote some ideas for changes, but nothing that really blocks merge imo.

Maybe the bit about the silent fail case could be changed but eh

**Comment by Sean1572** [comment]
wait seriously I can't just write a comment without it blocking the merge

are you for real? himm, okay if you are having trouble merging lmk i'll try to remove the comments on my side

File: .pylintrc (hunk 1 above)
Comment (sprestrelski) [reply .pylintrc:191]: e for exception

File: chunking_methods/WTS_chunking.py (hunk 4 above)
Comment (sprestrelski) [reply chunking_methods/WTS_chunking.py:1]: not necessarily :)

File: chunking_methods/gen_csv_labels.py (hunk 6 above)
Comment (sprestrelski) [reply chunking_methods/gen_csv_labels.py]: maybe not a runtime error, added a print statement for debug purposes but this is only called to prep for generating labels

File: chunking_methods/gen_csv_labels.py (hunk 5 above)
Comment (sprestrelski) [reply chunking_methods/gen_csv_labels.py]: started doing this, but realized I might need to rewrite a bit of it so these functions can be called perhaps from other files. everything is basically a global rn and that's not ideal

**Comment by sprestrelski** [comment]
not ready to merge yet - going to try a config file, but if it's too messy will leave as is

File: chunking_methods/gen_csv_labels.py (hunk 5 above)
Comment (sprestrelski) [reply chunking_methods/gen_csv_labels.py]: rewrite done!

**Comment by benjamin-cates** [comment]
Missing pydub dependency in PyHa environment

**Review by benjamin-cates — DISMISSED** [review]
Works

**Review by Sean1572 — APPROVED** [review]

**Review by benjamin-cates — APPROVED** [review]
